dali/test/python/test_utils.py

```diff
@@ -138,7 +181,12 @@ def compare_pipelines(pipe1, pipe2, batch_size, N_iterations, eps = 1e-07):
         for i in range(len(out1)):
             out1_data = out1[i].as_cpu() if isinstance(out1[i][0], dali.backend_impl.TensorGPU) else out1[i]
             out2_data = out2[i].as_cpu() if isinstance(out2[i][0], dali.backend_impl.TensorGPU) else out2[i]
-            check_batch(out1_data, out2_data, batch_size, eps)
+            if isinstance(expected_layout, tuple):
```

Comment:
list also?

Reply:
Yeah, it would be nice.

Reply:
I would say it's a test util, you need a iterable, you can extend it to your liking, but I don't see why we need to support 3 different possibilities here (as none is currently used).